module/VuFind/src/VuFind/Controller/AbstractBase.php

```diff
@@ -736,16 +735,44 @@ protected function normalizeUrlForComparison($url)
         return trim(end($parts), '/');
     }
 
+    /**
+     * Checks if a followup url is set
+     *
+     * @return bool
+     */
+    protected function hasFollowupUrl()
+    {
+        return null !== $this->followup()->retrieve('url');
+    }
+
     /**
      * Retrieve a referer to keep post-login redirect pointing
      * to an appropriate location.
      * Unset the followup before returning.
      *
+     * @param bool $checkRedirect Whether the query should be checked for param 'redirect'
+     *
      * @return string
      */
-    protected function getFollowupUrl()
-    {
-        return $this->followup()->retrieve('url', '');
+    protected function getAndClearFollowupUrl($checkRedirect = false)
+    {
+        if ($url = $this->followup()->retrieveAndClear('url', '')) {
+            $lightboxParent = $this->followup()->retrieveAndClear('lightboxParent');
+            // If a user clicks on the "Your Account" link, we want to be sure
+            // they get to their account rather than being redirected to an old
+            // followup URL. We'll use a redirect=0 GET flag to indicate this:
+            if (!$checkRedirect || $this->params()->fromQuery('redirect', true)) {
+                if (isset($lightboxParent) && $this->getAuthManager()->getSessionInitiator($this->getServerUrl())) {
+                    $this->followup()->store([], $url);
+                    $url = new \Laminas\Uri\Uri($lightboxParent);
+                    $params = $url->getQueryAsArray();
+                    $params['lightboxChild'] = urlencode($this->getServerUrl());
```

Comment:
Why is this double-encoded? The `urldecode` calls would also be unnecessary without the `urlencode` here.

Reply:
You are right. I was not aware of the url already being encoded, when its added this way. Removed it and the urldecode.